packages/components/search/src/search.js

```diff
@@ -52,6 +52,14 @@ customElementDefineHelper(
 					this.dispatchIdleEvent = helpers.debounceEvent(() => this.dispatchCustomEvent('idle', this.value), newVal);
 					break;
 				case 'value':
+					if (!this.hasUpdated) {
```

Comment:
I thought the problem was the debounce function, wasn't it?

Reply:
No, after debug I found that the problem happens because the debounce function is triggered on component initialisation (when initial value changed from `null` to `''`). This should be fixed by first check.
Second check tries to solve another problem, which is related to React re-render on a state update. I created the small example: https://codesandbox.io/s/react-wc-search-component-91hfs

Reply:
For the initial value check is it happening if developer set something or it's happening after we set it in the constructor?

Reply:
It happens in both cases, the difference only in the initial value in the `newVal` - empty string(which is came from the constructor) or the value from developer.